File: SuperMovie_server/src/api/Order/GetAllOrderWithUserId.cs
namespace SuperMovie.Server.Api.Order;

using WebSocketSharp;
using WebSocketSharp.Server;

public struct GetAllOrderWithUserIdReq
{
    public long OrderUserId;
}

public struct GetAllOrderWithUserIdRsp
{
    public List<OrderRsp> Collection;
}

//api : get_all_order_with_user
public class GetAllOrderWithUserId : WebSocketBehavior
{
}

File: movieui/src/scripts/ws/Order/getAllOrderWithUserId.ts
import {OrderRsp} from "@/scripts/ws/Order/getAllOrder"
import {wsUrlRoot} from "@/scripts/ws/meta"
import {createWebSocket, recvMsg, reqStringify, rspParse, sendMsg} from "@/scripts/ws/helper"

export {getAllOrderWithUserId}
export type {
    GetAllOrderWithUserIdReq,
    GetAllOrderWithUserIdRsp
}

type GetAllOrderWithUserIdReq =
    {
        OrderUserId: bigint
    }

type GetAllOrderWithUserIdRsp =
    {
        Collection: OrderRsp[]
    }

async function getAllOrderWithUserId(req: GetAllOrderWithUserIdReq) {
    const conn = createWebSocket(`${wsUrlRoot}/get_all_order_with_user_id`)

    const task = recvMsg(conn)
    const reqJson = reqStringify(req)
    console.log('get_all_order_with_user_id req:' + reqJson)
    sendMsg(conn, reqJson).then()
    const msg = await task
    console.log('get_all_order_with_user_id rsp:' + msg)

    return <GetAllOrderWithUserIdRsp>rspParse(msg)
}